apps/src/tutorialExplorer/responsive.jsx

```diff
@@ -0,0 +1,96 @@
+/* A very simple responsive layout system.
+ */
+
+/**
+ * Gets the container width.
+ * Returns either a number (e.g. 1170) or a string (e.g. "97%").
+ */
+
+function getResponsiveContainerWidth() {
+  const windowWidth = $(window).width();
+
+  if (windowWidth >= 1200) {
+    return 1170;
+  } else {
+    return "97%";
+  }
+}
+
+/* Window widths that are the starting points for each width category. */
+const responsiveWindowWidth = {
+  lg: 1024,
+  md: 820,
+  sm: 650,
+  xs: 0
+};
+
+/**
+ * Returns the window width that is the starting point for a width category.
+ *
+ * @param {string} id - "xs", "sm", "md", or "lg"
+ */
+
+function getResponsiveWindowWidth(category) {
```

Comment:
I wonder if this would be slightly easier to follow if we did
```
function fitsResponsiveCategory(category, windowWidth) {
  return windowWidth >= responsiveWindowWidth[category];
}
```
then elsewhere we have
```
fitsResponsiveCategory("lg", windowWidth)
```

Alternatively, you could do:
```
function getResponsiveCategory(windowWidth) {
  if (windowWidth >= responsiveWindowWidth["lg"]) {
    return "lg"
 }
 // etc. for rest of the options
}
```
Then in getResponsiveValue()
```
const category = getResponsiveCategory(windowWidth());
if (category === "lg") {
 ...
}  else if (category === "sm") {
 ...
}
// etc
```

Any of these approaches (including yours) are probably fine, though I might like the last one best.